### PR TITLE
Odb: user derating relocation

### PR DESCRIPTION
**Pull request by gudeh**
Bringing back this modification, since it showed necessary. RUDY requires some way to load the user derating without overwriting the GRT information, now that it is shared. For that we opted to put the user layer adjustment in ODB, in dbTechLayer. The code generator was used to do so.

### CODE REVIEW

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -1202,8 +1201,8 @@ void GlobalRouter::computeUserGlobalAdjustments(int min_routing_layer,
     return;
 
   for (int l = min_routing_layer; l <= max_routing_layer; l++) {
-    if (adjustments_[l] == 0) {
-      adjustments_[l] = adjustment_;
+    if (db_->getTech()->findRoutingLayer(l)->getLayerAdjustment() == 0.0) {
```
Comment (eder-matheus) [inline src/grt/src/GlobalRouter.cpp:1204]: Refactor this long function calls:
```
odb::dbTechLayer* tech_layer = getTech()->findRoutingLayer(l);
tech_layer->getLayerAdjustment()
```

And use the tech_layer in the lines below.

File: src/grt/src/GlobalRouter.cpp [hunk 2]
```diff
@@ -1218,7 +1217,8 @@ void GlobalRouter::computeUserLayerAdjustments(int max_routing_layer)
   const std::vector<int>& ver_capacities = grid_->getVerticalEdgesCapacities();
 
   for (int layer = 1; layer <= max_routing_layer; layer++) {
-    float adjustment = adjustments_[layer];
+    float adjustment
```
Comment (eder-matheus) [inline src/grt/src/GlobalRouter.cpp:1220]: Same here. Even if you only use it once, I think it's good to separate it for readability.

**Review by eder-matheus — CHANGES_REQUESTED** [review]

**Comment by rovinski** [comment]
Would be a good idea to rebase this PR to avoid the spurious changes.

**Comment by gudeh** [comment]
Closed due to too many unnecessary commits and failed rebase.

**Pull request closed without merging**